src/util/address.rs

```diff
@@ -77,29 +77,37 @@ pub enum Error {
     InvalidWitnessProgramLength(usize),
     /// A v0 witness program must be either of length 20 or 32.
     InvalidSegwitV0ProgramLength(usize),
+    /// A v1 witness program (taproot) must be either of length 20 or 32.
+    InvalidTaprootProgramLength(usize),
     /// An uncompressed pubkey was used where it is not allowed.
     UncompressedPubkey,
+    /// Witness version which is not (yet) supported
+    UnsupportedWitnessVersion(WitnessVersion),
 }
 
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
-            Error::Base58(_) => write!(f, "base58 address encoding error"),
-            Error::Bech32(_) => write!(f, "bech32 address encoding error"),
-            Error::EmptyBech32Payload => write!(f, "the bech32 payload was empty"),
+            Error::Base58(_) => f.write_str("base58 address encoding error"),
+            Error::Bech32(_) => f.write_str( "bech32 address encoding error"),
+            Error::EmptyBech32Payload => f.write_str( "the bech32 payload was empty"),
             Error::InvalidBech32Variant { expected, found } => write!(f, "invalid bech32 checksum variant found {:?} when {:?} was expected", found, expected),
             Error::InvalidWitnessVersion(v) => write!(f, "invalid witness script version: {}", v),
-            Error::UnparsableWitnessVersion(_) => write!(f, "incorrect format of a witness version byte"),
+            Error::UnparsableWitnessVersion(_) => f.write_str( "incorrect format of a witness version byte"),
             Error::MalformedWitnessVersion => f.write_str("bitcoin script opcode does not match any known witness version, the script is malformed"),
             Error::InvalidWitnessProgramLength(l) => write!(f,
                 "the witness program must be between 2 and 40 bytes in length: length={}", l,
             ),
             Error::InvalidSegwitV0ProgramLength(l) => write!(f,
                 "a v0 witness program must be either of length 20 or 32 bytes: length={}", l,
             ),
-            Error::UncompressedPubkey => write!(f,
+            Error::InvalidTaprootProgramLength(l) => write!(f,
+                "a v1 witness program (taproot) must be either of length 20 or 32 bytes: length={}", l,
```

Comment:
This `l` looks a bit like `1` so would be nice to use `len` instead.

Reply:
Same length error here. v1 witness programs are always 32 bytes

Reply:
The len is what was given, not what is expected.